crates/cast/src/helpers/scarb_utils.rs

```diff
@@ -69,18 +70,47 @@ pub fn get_scarb_manifest() -> Result<Utf8PathBuf> {
     Ok(path)
 }
 
-pub fn get_scarb_metadata(manifest_path: &Utf8PathBuf) -> Result<scarb_metadata::Metadata> {
+pub fn get_scarb_metadata(
+    manifest_path: &Utf8PathBuf,
+    with_deps: bool,
+) -> Result<scarb_metadata::Metadata> {
     which::which("scarb")
         .context("Cannot find `scarb` binary in PATH. Make sure you have Scarb installed https://github.com/software-mansion/scarb")?;
 
-    scarb_metadata::MetadataCommand::new()
-        .inherit_stderr()
-        .manifest_path(manifest_path)
-        .no_deps()
-        .exec()
-        .context(
-            format!("Failed to read Scarb.toml manifest file, not found in current nor parent directories, {}", env::current_dir().unwrap().into_os_string().into_string().unwrap()),
-        )
+    let mut binding = scarb_metadata::MetadataCommand::new();
+    let mut command = binding.inherit_stderr().manifest_path(manifest_path);
+
+    if !with_deps {
+        command = command.no_deps();
+    }
+
+    command.exec().context(format!(
+        "Failed to read Scarb.toml manifest file, not found in current nor parent directories, {}",
+        env::current_dir()
+            .unwrap()
+            .into_os_string()
+            .into_string()
+            .unwrap()
+    ))
+}
+
+pub fn parse_scarb_metadata(path: &Option<Utf8PathBuf>) -> Result<Metadata> {
+    if let Some(path) = path {
+        if !path.exists() {
+            bail!("{path} file does not exist!");
+        }
+    }
+
+    let manifest_path = match path.clone() {
+        Some(path) => path,
+        None => get_scarb_manifest().context("Failed to obtain manifest path from scarb")?,
+    };
```

Comment:
Could these be combined? Quite weird to split this handling to me

Reply:
updated